showyourwork/config.py

```diff
@@ -401,6 +398,17 @@ def parse_config():
     # Git info for the repo
     config["git_sha"] = git.get_repo_sha()
     config["git_url"] = git.get_repo_url()
+    config["git_short_url"] = (
+        config["git_url"]
+        .replace("https://", "")
+        .replace("http://", "")
+        .replace("_", "-")
```

Comment:
Would it be better to stick it inside a `\url{}` instead of changing to `-`?

Reply:
Definitely, or any of `\_`, `$_$`, `\char`_`... But I'm using a "path decoration" in TikZ (see [here](https://github.com/rodluger/showyourwork-dev/blob/watermark/showyourwork/workflow/resources/styles/build.tex#L88-L95)), which appears to only allow plain text with very limited formatting and no special characters. There's no error if you include a special character, but the build just hangs indefinitely. I think this is probably a bug in TikZ, but I haven't been able to track anything down online. This is also, by the way, the reason I wasn't able to include the GitHub logo along the path.

Reply:
The best approach here would probably be to abandon this "path decoration" approach entirely, but I don't know how else to generate curved text like that. There may be a different way of doing it with TikZ, but I'm not sure.

Reply:
Ha, the URL in the hyperlink itself is fine -- that doesn't get changed, and the link points to the correct repo. It just _displays_ as a hyphen. I hate it, but I currently don't have an alternative.

Reply:
Ah! [This](https://tex.stackexchange.com/a/138730) is why that's happening: TikZ places one character at a time along the path, so things like `$`, `\`, etc., which are parts of multi-character expressions, cause it to fail. Let me look into this more -- seems like that thread had a solution.

Reply:
Looks like we just need to do wrap the thing in curly braces. I'll work on it after lunch!

Reply:
OK, trick was to replace "_" with "{\_}". Working now!